src/main.rs

```diff
@@ -14,9 +14,37 @@
  * limitations under the License.
  */
 
+use std::sync::Arc;
+
 use quilkin::runner::run;
 
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+
 #[tokio::main]
 async fn main() -> Result<(), quilkin::runner::Error> {
-    run(vec![]).await
+    let log = quilkin::proxy::logger();
+    let version: std::borrow::Cow<'static, str> = if cfg!(debug_assertions) {
+        format!("{}+debug", VERSION).into()
+    } else {
+        VERSION.into()
+    };
+
+    slog::info!(log, "Starting Quilkin"; "version" => &*version);
+
+    let matches = clap::App::new(clap::crate_name!())
+        .version(&*version)
+        .about(clap::crate_description!())
+        .arg(
+            clap::Arg::with_name("config")
+                .short("c")
+                .long("config")
+                .value_name("CONFIG")
+                .help("The YAML configuration file")
+                .takes_value(true),
+        )
+        .get_matches();
+
+    let config = quilkin::config::Config::find(&log, matches.value_of("config")).map(Arc::new)?;
+
+    run(log, config, vec![]).await
```

Comment:
I like this a lot, as then if people desire to have alternate config builders, they are able to do so. 👍🏻